File: Algorithm/1-TwoPointers/283_moveZeros.java
package LeetCode.Array;

class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length;
        int slow = 0, fast = 0;
        while (fast < n) {
            if (nums[fast] != 0) {
                nums[slow] = nums[fast];
                slow++;
            }
            fast++;
        }
        while (slow < n) {
            nums[slow] = 0;
            slow++;
        }
    }
}

File: Algorithm/1-TwoPointers/283_moveZeros.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int cnt = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[cnt] == 0 && nums[i] != 0) {
                swap(nums[cnt], nums[i]);
                cnt++;
            } else if (nums[cnt] != 0)
                cnt++;
        }
    }
};
// 通过了 思路还可以

class Answer {
public: 
    void moveZeros(vector<int>& nums) {
        for (int cnt = 0, cur = 0; cur < nums.size(); cur++) {
            if (nums[cur] != 0)
                swap(nums[cnt++], nums[cur]);
        }
    }
};
// 有很多自己跟自己交换的